lib/util/fs.js

```diff
@@ -31,11 +31,42 @@ function ensureDirectoryExists(dir) {
   return stat(dir).catch(() => mkdirp(dir));
 }
 
+/**
+ * Read recursively a directory,
+ * sorting folders and files by alphabetically order
+ *
+ * @param {string} dir
+ * The directory to recursively analyse
+ */
+async function getFilepathsInFolderRecursively(dir) {
+  let results = [];
+  try {
+    const pathsList = await readdir(dir);
+    await Promise.all(
+      pathsList.sort().map(async (currentPath) => {
+        const currentFile = path.resolve(dir, currentPath);
+        const statFile = await stat(currentFile);
+        if (statFile && statFile.isDirectory()) {
+          // Add directory to array [comment if you need to remove the directories from the array]
+          const subFiles = await getFilepathsInFolderRecursively(currentFile);
+          results = results.concat(subFiles);
+        } else {
+          results.push(currentFile);
+        }
+      })
+    );
+  } catch (error) {
+    throw new Error(error);
```

Comment:
why catch it in the first place?

Reply:
the use of `readdir` and `stat` make me think it could break at the beginning,
do you have an other idea ?

Reply:
but even if it breaks in the beginning, what do we gain from immediately rethrowing an error wrapped into generic Error?